### DISCUSSION TITLE
reference to conditional fields are not resolve properly (maybe?)

### DISCUSSION THREAD

**Discussion by wylswz** [cue-lang/cue#2563] [Q&A] [answered]
I wrote following code

```
a : *[] | [...]

result: {

    if true {
      a: [1,2,3]
    }
    
    res: a
}
```

and I got this 
```
a: []
result: {
	res: []
	a: [1, 2, 3]
}

```

instead of 

```
a: []
result: {
	res: [1, 2, 3]
	a: [1, 2, 3]
}

```

Just wondering what's happening here, is it by design?

**Comment by myitcv** [comment, marked as answer]
CUE is [lexically scoped using blocks](https://tip.cuelang.org/docs/references/spec/#declarations-and-scope), which means that the reference to `a` in `res: a` resolves to the top-level `a : *[] | [...]` and not the one contained within the if comprehension. 

Depends what you are after here: https://tip.cuelang.org/play/?id=nbnzUOsJafS#cue@export@cue

**Reply by wylswz** [reply]
Thanks for your reply, that exactly answered my question.

**Comment by wylswz** [comment]
It's more clear if I remove the `if true`, then `{a: [1,2,3]}` becomes a embedded struct, which cannot be referred by it's neighbour `res`
